javalib/src/main/scala/java/math/BigInteger.scala

```diff
@@ -0,0 +1,1500 @@
+   * <p>
+   * <b>Implementation Note:</b> Currently {@code rnd} is ignored.
+   *
+   * @param bitLength length of the new {@code BigInteger} in bits.
+   * @param rnd random generator used to generate the new {@code BigInteger}.
+   * @return probably prime random {@code BigInteger} instance.
+   * @throws ArithmeticException if {@code bitLength < 2}.
+   */
+  def probablePrime(bitLength: Int, rnd: Random): BigInteger = new BigInteger(bitLength, 100, rnd)
+
+  def valueOf(lVal: Long): BigInteger = {
+    if (lVal < 0) {
+      if (lVal != -1) {
+        new BigInteger(-1, -lVal)
+      } else {
+        MINUS_ONE
+      }
+    } else if (lVal <= 10) {
+      SMALL_VALUES(lVal.toInt)
+    } else {
+      new BigInteger(1, lVal)
+    }
+  }
+
+  def getPowerOfTwo(exp: Int): BigInteger = {
+    if (exp < TWO_POWS.length) {
+      TWO_POWS(exp)
+    } else {
+      val intCount = exp >> 5
+      val bitN = exp & 31
+      val resDigits = Array.ofDim[Int](intCount + 1)
+      resDigits(intCount) = 1 << bitN
+      new BigInteger(1, intCount + 1, resDigits)
+    }
+  }
+
+  private def checkNotNull[T](reference: T): T = {
+    if (reference == null) {
+      throw new NullPointerException()
+    }
+    reference
+  }
+
+  private def checkCriticalArgument(expression: Boolean, errorMessage: AnyRef) {
+    if (!expression) {
+      throw new IllegalArgumentException(String.valueOf(errorMessage))
+    }
+  }
+}
+
+/**
+ * This class represents immutable integer numbers of arbitrary length. Large
+ * numbers are typically used in security applications and therefore BigIntegers
+ * offer dedicated functionality like the generation of large prime numbers or
+ * the computation of modular inverse.
+ * <p>
+ * Since the class was modeled to offer all the functionality as the
+ * {@link Integer} class does, it provides even methods that operate bitwise on
+ * a two's complement representation of large integers. Note however that the
+ * implementations favors an internal representation where magnitude and sign
+ * are treated separately. Hence such operations are inefficient and should be
+ * discouraged. In simple words: Do NOT implement any bit fields based on
+ * BigInteger.
+ */
+class BigInteger  extends Number with Comparable[BigInteger] with Serializable {
+  import BigInteger._
+
+  /**
+   * The magnitude of this big integer. This array is in little endian order and
+   * each "digit" is a 32-bit unsigned integer. For example: {@code 13} is
+   * represented as [ 13 ] {@code -13} is represented as [ 13 ] {@code 2^32 +
+   * 13} is represented as [ 13, 1 ] {@code 2^64 + 13} is represented as [ 13,
+   * 0, 1 ] {@code 2^31} is represented as [ Integer.MIN_VALUE ] The magnitude
+   * array may be longer than strictly necessary, which results in additional
+   * trailing zeros.
+   *
+   * <p>TODO(jat): consider changing to 24-bit integers for better performance
+   * in browsers.
+   */
+  @transient var digits: Array[Int] = _
```

Comment:
Do not use `@transient` in Scala.js. It is irrelevant.
Make this `private`.
Same for the other variables herebelow.